### PR TITLE
report edgar/ez-uitreemenu-bundle to ezsystems/ezplatform-admin-ui

### PR DESCRIPTION
**Pull request by noodle69**
| Question      | Answer
| ------------- | ---
| Tickets       | 
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | no
| Doc needed?   | yes
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

Integrate new Browse tree feature at content left sidebar.
When i tried to extract translation, i have some issues : there is only to string to translate "Browse tree" button label and widget title.

#### Checklist:
- [ ] Coding standards (`$ composer fix-cs`)
- [ ] Ready for Code Review

### CODE REVIEW

**Comment by SylvainGuittard** [comment]
Thanks @noodle69 for this great contribution.
@sunpietro: about the icon, there is no svg icon for the content tree so far. @inakijv will have to create one.

**Comment by sunpietro** [comment]
@noodle69 @SylvainGuittard definitely good idea to introduce such a functionality in eZ Platform.
I have a suggestion. Functionality of the content tree is similar as Universal Discovery (the browse tab), the only difference is a design. My proposition is to reuse as much as you can from Universal Discovery (with some tweaks of course) and make it a ReactJS module (for better rendering performance).

**Comment by noodle69** [comment]
@sunpietro @SylvainGuittard 
i've fix php cs and commit routing with "Notification" routes previously added

i start a new mission start of July, i will not have time to convert css to scss.

**Comment by noodle69** [comment]
@sunpietro 
I started on my repo a branch to integrate this browse tree on UDW
I let you discuss with eZ Team if button "Browse tree" at left content sidebar is pertinent
It's just a proposal

File: src/bundle/Controller/BrowseTreeController.php [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+     * @return Response
+     */
+    public function childrenAction(Location $location, int $offset = 0): Response
+    {
+        $response = new JsonResponse();
+
+        try {
+            $children = $this->findNodes($location, null, $offset);
+        } catch (NotFoundException $e) {
+            $children = null;
```
Comment (Plopix) [inline src/bundle/Controller/BrowseTreeController.php]:
```php
$children = [];
```

File: src/bundle/Controller/BrowseTreeController.php [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+            if ($this->loadLocationChildrenCount($child)) {
+                $nodesTemp = new Node();
+                $nodesTemp->text = '...';
+                $childNode->children = [$nodesTemp];
+            }
+
+            $nodes[] = $childNode;
+        }
+
+        if (empty($nodes)) {
```
Comment (Plopix) [inline src/bundle/Controller/BrowseTreeController.php]: Remove that if (it is not useful)

File: src/bundle/Controller/BrowseTreeController.php [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+     * @param Location  $location
+     * @param Node|null $node
+     * @param int       $offset
+     * @param bool      $init
+     *
+     * @return array|null
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    protected function findNodes(Location $location, ?Node $node = null, int $offset = 0, bool $init = false): ?array
```
Comment (Plopix) [inline src/bundle/Controller/BrowseTreeController.php]: I would rename that too: findNodes -> findLocations ?

Comment (Plopix) [reply src/bundle/Controller/BrowseTreeController.php]: And all the `$node` references

**Comment by inakijv** [comment]
Content tree icon requested.

[content-tree.svg.zip](https://github.com/ezsystems/ezplatform-admin-ui/files/2158771/content-tree.svg.zip)

**Comment by lserwatka** [comment]
Closing in favour of Content Tree implementation in the eZ Platform 2.5 LTS.

**Pull request closed without merging**